### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by bojand** [form-data/form-data#175]
Hello, while investigating [mailgun-js issue #90](https://github.com/1lobby/mailgun-js/issues/90), I think I may be hitting a bug in the `form-data` module, or `combined-stream` possibly; not sure. Here's a simpler example:

```
// server
var express = require('express');
var app = express();
var port = 3000;

app.post('/api_route', function(req, res) {
  console.log('api route');
  res.send('done');
});

app.listen(port, function() {
  console.log('Example app listening on port 3000!');
});


// client side
var FormData = require('form-data');

var form = new FormData();
var headers;

function getDataValue(key, input) {
  if (typeof input === 'object') {
    return JSON.stringify(input);
  } else if (typeof input === 'number' || typeof input === 'boolean') {
    return input.toString();
  } else {
    return input;
  }
}

function prepareFormData(data) {
  for (var key in data) {
    var obj = data[key];
    if (Array.isArray(obj)) {
      function appendKey(element) {
        form.append(key, getDataValue(key, element));
      }

      obj.forEach(appendKey);
    } else {
      form.append(key, getDataValue(key, obj));
    }
  }

  headers = form.getHeaders();
};

var to = [];
var N = 999;

for (var i = 0; i <= N; i++) {
  to.push('mailgunjs' + '+' + i + '@gmail.com');
}

var p = {
  to: to,
  from: 'Test user <test@acme.com>',
  subject: 'Hey %recipient.first%',
  text: 'testing message for mailgun for %recipient.first%'
}

prepareFormData(p);

var opts = {
  hostname: 'localhost',
  port: port,
  protocol: 'http:',
  path: '/api_route',
  method: 'POST',
  headers: headers
};

form.submit(opts, function(err, res) {
  if (err) console.log(err);
  console.log('status code: %s', res.statusCode);
  process.exit(0);
});
```

```
npm i express form-data
node index.js
```
output:
```
Example app listening on port 3000!

(null):0
(null)

RangeError: Maximum call stack size exceeded
```

Interesting thing of note is that if I change the `p` parameters object, the maximum number of items `N` changes before I start getting the error. Sometimes I can do more, sometimes I can do less. For example I've gotten it to crash with < 500 items too. 

Any advice on how to further debug would be appreciated, or what to check; and if I could fix it I'd be happy to provide a PR. Will try and look further into it.

Thanks,

Bojan

**Comment by factorial** [comment]
I am having the same issue as @bojand in the same file but my problem is happening during construction of a request object from the request module, a package which also uses form-data module.

**Comment by alexindigo** [comment]
Yeah, looks like it's async.parallel's doing. And there is somewhat related PR https://github.com/form-data/form-data/pull/204 – although it doesn't this specific problem. But we're in the works to fix/improve this part of the code.

**Comment by alexindigo** [comment]
Should be fixed in 2.0.0 Thank you.
